README.md

```diff
@@ -75,6 +75,17 @@ The following tasks are available in Hardhat:
    - `--uni-proxy`: The address of the Gamma UniProxy contract used for deploying liquidity.
    - `--solid-staking`: The address of our SolidStaking contract, to stake with recipient.
 
+5. Deploy a new SolidZapDecollateralize contract:
+   ```shell
+   yarn hardhat --network localhost deploy-zap-decollateralize [OPTIONS]
+   ```
+
+   Options:
+   - `--router`: The address of the router used to perform token swaps.
```

Comment:
I was thinking if router needs to be hardcoded. Can we just pass it as an argument with encoded swap?

Reply:
what we ultimately need is to do `router.call(encodedSwap);`

we can encode the address of the router into the swap itself on the frontend, e.g. the encoded swap would have 2 parts, (address router ; bytes actualSwap)

then we would do an on-chain `abi.decode` to obtain the 2 parts, and call `router.call(actualSwap);`

but we know the router from the beginning, it's simple to have it during contract deployment as constructor argument. it also enables `IWETH(weth).approve(_router, type(uint).max);` in the constructor

Reply:
What if we decide to support another router? Or allow users to decide what router to use.

Reply:
We deploy another zap with different router

Reply:
Cause if the router is different, we might also need to adjust what we do after `router.call(encodedSwap);`. 
Kyberswap router for example returns amount swapped from `router.call(encodedSwap)`, and we can process it